src/Database/PostgreSQL/Consumers/Components.hs

```diff
@@ -75,13 +75,14 @@ runConsumerWithMaybeIdleSignal cc cs mIdleSignal
       skipLockedTest :: Either DBException () <-
         try . runDBT cs defaultTransactionSettings $
         runSQL_ "SELECT TRUE FOR UPDATE SKIP LOCKED"
-      let useSkipLocked = either (const False) (const True) skipLockedTest
+      -- If we can't lock rows using 'skip locked' throw an exception
+      either throwM pure skipLockedTest
```

Comment:
Rethrowing `DBException` is a bit indirect, we can just `error "PostgreSQL version with support for SKIP LOCKED is required"` or something like that, it'll be more informative.